### PR TITLE
Job Queue Push/Pop/Fail/Success commands

### PR DESCRIPTION
**Pull request by gshilin-sdb**

### CODE REVIEW

**Comment by jhernand** [comment]
Looks good to me.

**Review by igoihman — CHANGES_REQUESTED** [review]
We moved to commands structure of `ocm <action> <resource>`. Can you change the implementation so it will adhere to the new structure?

**Comment by gshilin-sdb** [comment]
Action is `jobs` and resource is `fail/pop/push/success`. All of them are POST. I'd prefer not to make it `post jobs push <parameters>`, but to keep it short and concise: `jobs push <parameters>`.

**Comment by igoihman** [comment]
@gshilin-sdb the commands should be `ocm push job`. `post` shouldn't be part of it

**Comment by gshilin-sdb** [comment]
refactored

File: cmd/ocm/fail/job/job.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift-online/ocm-cli/pkg/ocm"
+)
+
+// Cmd Constant:
+var Cmd = &cobra.Command{
+	Use:     "failure QUEUE_NAME JOB_ID RECEIPT_ID FAILURE_REASON",
```
Comment (bvulaj) [inline cmd/ocm/fail/job/job.go:29]: Is this accurate?  The resource is `fail`, so it should be `fail job QUEUE_NAME...` not `fail failure QUEUE_NAME`

I think you have this correct for the success resource.

File: cmd/ocm/main.go [hunk 2]
```diff
@@ -95,6 +99,11 @@ func init() {
 	root.AddCommand(plugincmd.Cmd)
 	root.AddCommand(hibernate.Cmd)
 	root.AddCommand(resume.Cmd)
+	root.AddCommand(post.Cmd)
```
Comment (bvulaj) [inline cmd/ocm/main.go:102]: Unnecessary.

**Review by bvulaj — CHANGES_REQUESTED** [review]

File: cmd/ocm/fail/job/job.go (hunk 1 above)
Comment (gshilin-sdb) [reply cmd/ocm/fail/job/job.go:29]: Fixed

File: cmd/ocm/main.go (hunk 2 above)
Comment (gshilin-sdb) [reply cmd/ocm/main.go:102]: removed

**Comment by gshilin-sdb** [comment]
All fixed. @igoihman PTAL

File: cmd/ocm/fail/job/job.go (hunk 1 above)
Comment (gshilin-sdb) [reply cmd/ocm/fail/job/job.go]: done

File: cmd/ocm/fail/job/job.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	// Get the client for the Job Queue management api
+	client := connection.JobQueue().V1()
+
+	// Send a request to Success a Job:
+	_, err = client.Queues().Queue(argv[0]).Jobs().Job(argv[1]).Failure().ReceiptId(argv[2]).FailureReason(argv[3]).Send()
```
Comment (igoihman) [inline cmd/ocm/fail/job/job.go:50]: missing validation for argv items existance before calling them

File: cmd/ocm/pop/job/job.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	// Get the client for the Job Queue management api
+	client := connection.JobQueue().V1()
+
+	// Send a request to create the Job:
+	pop, err = client.Queues().Queue(argv[0]).Pop().Send()
```
Comment (igoihman) [inline cmd/ocm/pop/job/job.go:54]: missing validation for argv

File: cmd/ocm/push/job/job.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	// Get the client for the Job Queue management api
+	client := connection.JobQueue().V1()
+
+	// Send a request to create a Job:
+	request := client.Queues().Queue(argv[0]).Push()
```
Comment (igoihman) [inline cmd/ocm/push/job/job.go:66]: same

File: cmd/ocm/success/job/job.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	// Get the client for the Job Queue management api
+	client := connection.JobQueue().V1()
+
+	// Send a request to Success a Job:
+	_, err = client.Queues().Queue(argv[0]).Jobs().Job(argv[1]).Success().ReceiptId(argv[2]).Send()
```
Comment (igoihman) [inline cmd/ocm/success/job/job.go:48]: same

File: cmd/ocm/success/cmd.go [hunk 7]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
Comment (igoihman) [inline cmd/ocm/success/cmd.go:2]: 2021

File: cmd/ocm/push/cmd.go [hunk 8]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
Comment (igoihman) [inline cmd/ocm/push/cmd.go:2]: 2021

File: cmd/ocm/pop/cmd.go [hunk 9]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
Comment (igoihman) [inline cmd/ocm/pop/cmd.go:2]: 2021

File: cmd/ocm/fail/cmd.go [hunk 10]
```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
```
Comment (igoihman) [inline cmd/ocm/fail/cmd.go:2]: 2021

File: cmd/ocm/fail/job/job.go (hunk 3 above)
Comment (gshilin-sdb) [reply cmd/ocm/fail/job/job.go:50]: There is no way to validate argv items, but to count them, so there is `cobra.ExactArgs(4)`

File: cmd/ocm/pop/job/job.go (hunk 4 above)
Comment (gshilin-sdb) [reply cmd/ocm/pop/job/job.go:54]: `cobra.ExactArgs(1)`

File: cmd/ocm/push/job/job.go (hunk 5 above)
Comment (gshilin-sdb) [reply cmd/ocm/push/job/job.go:66]: `cobra.ExactArgs(1)`

File: cmd/ocm/success/job/job.go (hunk 6 above)
Comment (gshilin-sdb) [reply cmd/ocm/success/job/job.go:48]: `cobra.ExactArgs(3)`

File: cmd/ocm/success/cmd.go (hunk 7 above)
Comment (gshilin-sdb) [reply cmd/ocm/success/cmd.go:2]: fixed

File: cmd/ocm/push/cmd.go (hunk 8 above)
Comment (gshilin-sdb) [reply cmd/ocm/push/cmd.go:2]: fixed

File: cmd/ocm/pop/cmd.go (hunk 9 above)
Comment (gshilin-sdb) [reply cmd/ocm/pop/cmd.go:2]: fixed

File: cmd/ocm/fail/cmd.go (hunk 10 above)
Comment (gshilin-sdb) [reply cmd/ocm/fail/cmd.go:2]: fixed

**Comment by gshilin-sdb** [comment]
All commented/fixed. @igoihman PTAL

**Review by igoihman — APPROVED** [review]
/lgtm

**Pull request merged**